### PR TITLE
Migrate target abort and activate

### PR DESCRIPTION
**Pull request by howbazaar**
Adds Abort and Activate to the MigrationTarget API facade and client.

### CODE REVIEW

File: state/state_test.go [hunk 1]
```diff
@@ -2952,6 +2956,24 @@ func (s *StateSuite) TestRemoveAllEnvironDocs(c *gc.C) {
 	}
 }
 
+func (s *StateSuite) TestRemoveAllEnvironDocs(c *gc.C) {
```
Comment (mjs) [inline state/state_test.go]: This should be called TestRemoveAllModelDocs

File: state/state_test.go [hunk 2]
```diff
@@ -2952,6 +2956,24 @@ func (s *StateSuite) TestRemoveAllEnvironDocs(c *gc.C) {
...
+
+	err = st.RemoveAllModelDocs()
+	c.Assert(err, jc.ErrorIsNil)
+
+	// test that we can not find the user:envName unique index
+	s.checkUserModelNameExists(c, checkUserModelNameArgs{st: st, id: userModelKey, exists: false})
+	s.AssertModelDeleted(c, st)
+}
+
 func (s *StateSuite) TestRemoveAllEnvironDocsAliveEnvFails(c *gc.C) {
```
Comment (mjs) [inline state/state_test.go]: This name is wrong too

File: api/migrationtarget/doc.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2016 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-// This package defines the client side API facade for use by the
+// Package migrationtarget defines the client side API facade for use by the
```
Comment (mjs) [inline api/migrationtarget/doc.go]: Ha! I've just learned that this is the "right" way to do package docstrings in Go.

**Comment by mjs** [comment]
LGTM

**Comment by howbazaar** [comment]
$$merge$$

**Comment by howbazaar** [comment]
$$resolved-conflicts$$

**Pull request merged**